compiler/modules/front.syntax/parse.bal

```diff
@@ -133,38 +137,53 @@ function parseModuleDecl(Tokenizer tok, ModulePart part) returns ModuleLevelDefn
 }
 
 function parseTypeDefinition(Tokenizer tok, ModulePart part, Visibility vis) returns TypeDefn|err:Syntax {
+    Position startPos = tok.currentPos();
+    if vis != () {
+        startPos = tok.previousPos();
+    }
     check tok.advance();
-    Position pos = tok.currentPos();
+    Position namePos = tok.currentPos();
     string name = check tok.expectIdentifier();
     TypeDesc td = check parseTypeDesc(tok);
+    Position endPos = tok.currentPos();
     check tok.expect(";");
-    return { name, td, pos, vis, part };
+    return { startPos, endPos, name, td, namePos, vis, part };
 }
 
 function parseConstDefinition(Tokenizer tok, ModulePart part, Visibility vis) returns ConstDefn|err:Syntax {
+    Position startPos = tok.currentPos();
+    if vis != () {
+        startPos = tok.previousPos();
+    }
     check tok.advance();
-    Position pos = tok.currentPos();
     Token? t = tok.current();
     InlineBuiltinTypeDesc? td = ();
     if t is InlineBuiltinTypeDesc {
         check tok.advance();
         td = t;
     }
+    Position namePos = tok.currentPos();
     string name = check tok.expectIdentifier();
     check tok.expect("=");
     Expr expr = check parseInnerExpr(tok);
+    Position endPos = tok.currentPos();
```

Comment:
We want the endPos to be the end of the `;` token. The way to do that is to have a function like `expect` that returns the endPos of the expected token.

Reply:
fixed